### PR TITLE
Settings Sync: Auto Archive

### PR DESCRIPTION
**Pull request by bjtitus**
| 📘 Part of: #1400 |
|:---:|

Adds Auto Archive syncing for General and Podcast settings.

https://github.com/Automattic/pocket-casts-ios/assets/3250/1eae1579-d2f4-4bfa-bd94-a340668f6e6c

## Changes

* Adds three new General settings `autoArchivePlayed`, `autoArchiveInactive`, and `autoArchiveIncludesStarred`

https://github.com/Automattic/pocket-casts-ios/blob/7330deb80bd9d77ae0d5a48859989458ffb3fcd0/Modules/Server/Sources/PocketCastsServer/Public/AppSettings.swift#L30-L31

https://github.com/Automattic/pocket-casts-ios/blob/50584cf40d04579a7fd490395e14cfcff1412ecb/Modules/Server/Sources/PocketCastsServer/Public/AppSettings.swift#L32

Adds server request and response handling + UserDefaults imports for each of these settings.

* Adds three new Podcast flags, `autoArchivePlayed`, `autoArchiveInactive`, `autoArchiveEpisodeLimit`

https://github.com/Automattic/pocket-casts-ios/blob/22081645ffc8addac609f7975c66764f6588f82c/Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/PodcastSettings.swift#L14-L16

* Adds `autoArchiveInactiveAfterTime`, `autoArchiveEpisodeLimitCount`, and `isAutoArchiveOverridden` settings wrappers to Podcast.

https://github.com/Automattic/pocket-casts-ios/blob/5415a8bf944324f27bb54fb11cc340f943d304b7/Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/Podcast%2BSettings.swift#L84-L117

https://github.com/Automattic/pocket-casts-ios/blob/0fbb15fb4f492a4096558cf368cfc96bb3453220/Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/Podcast%2BSettings.swift#L121-L136

* Sets `syncStatus` when any of the above Podcast settings wrappers have changed

## To test

> [!IMPORTANT]
> Must use Staging build configuration. Some of these settings are only in staging and not production.

#### Auto Archive General setting

1. Log in with the same account on two devices
2. D1: Navigate to Profile > Settings > Auto Archive
3. D1: Change some of the settings
4. D2: Pull to Refresh the Profile
5. D2: Check Profile > Settings > Auto Archive and ensure values match
6. D2: Play a podcast and ensure the setting applies at the end

#### Auto Archive Podcast setting

1. Log in with the same account on two devices
2. D1: Navigate to a Podcast's Settings page
3. D1: Enable "Custom for this Podcast"
4. D1: Change settings to apply differently than General settings (maybe enable for a single podcast)
5. D1: Pull to Refresh Podcasts list
6. D2: Pull to Refresh Podcasts list
7. D2: Check the Podcast's settings and ensure values match
8. D2: Play a podcast and ensure the setting applies at the end

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/41">7.58</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/Podcast+Settings.swift [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import PocketCastsUtils
+import Foundation
+
+extension Podcast {
+    public var isEffectsOverridden: Bool {
```
Comment (leandroalonso) [inline Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/Podcast+Settings.swift:5]: Should this be part of this PR?

I don't see any mention of per-podcast playback settings test steps. 🤔

**Review by leandroalonso — CHANGES_REQUESTED** [review]
@bjtitus I'm unable to sync per-podcast settings:

https://github.com/Automattic/pocket-casts-ios/assets/7040243/5b1391ba-e79b-4aa1-89de-cf5e7180d3c6

Also, it seems that there's also code for syncing playback effects?

File: Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/Podcast+Settings.swift (hunk 1 above)
Comment (bjtitus) [reply Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/Podcast+Settings.swift:5]: This file was moved in its entirety from the main project to the `DataModel` module. Maybe I need to use `git` to reclassify as a move?

Comment (bjtitus) [reply Modules/DataModel/Sources/PocketCastsDataModel/Public/Model/Podcast+Settings.swift:5]: For some reason, the `Podcast+Settings.swift` file from the podcasts directory was still around on-disk but not in the project I deleted it [here](https://github.com/Automattic/pocket-casts-ios/pull/1455/commits/81ec94ddcc573107d82fb3ef6754c13edc9ca478).

**Comment by bjtitus** [comment]
I apologize, something got borked with this branch and I must have overlooked it in checking the diff. This branch should be cleaned up and ready to go now.

The `autoArchive` override flag was missing and is added in https://github.com/Automattic/pocket-casts-ios/pull/1455/commits/319982afd6e8733ce7b079a952ece56eb7ac7da5 along with a settings wrapper, `isAutoArchiveOverridden` in https://github.com/Automattic/pocket-casts-ios/pull/1455/commits/0fbb15fb4f492a4096558cf368cfc96bb3453220.

The `syncStatus` also needed to be changed when these Auto Archive settings were changed, so I added that in the settings wrappers in https://github.com/Automattic/pocket-casts-ios/pull/1455/commits/caa95f009db49e1a0fef30689b0efa6f2f19b14b

Sorry for the large diff! I thought it was better to just include all of these Auto Archive settings at once so they can be tested and merged together.

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**